com.unity.shadergraph/Editor/Data/Interfaces/GenerationMode.cs

```diff
@@ -3,7 +3,8 @@ namespace UnityEditor.ShaderGraph
     enum GenerationMode
     {
         Preview,
-        ForReals
+        ForReals,
+        VFX
```

Comment:
Introduce a third Generation Mode specific for VFX. This is currently required because VFX must configure vertex/surface description functions per-instance.

Reply:
Yeah, Ideally we would want to find a way to make this more generic, so you can configure it all from the Target/Subtarget API level. but I think that's a more architectural task for ShaderGraph.  This should be fine for now.